### PR TITLE
XDR scan abort command

### PR DESCRIPTION
**Pull request by barchen1**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/34841
fixes: https://github.com/demisto/etc/issues/35517

## Description
1.Added the ***xdr-endpoint-scan-abort*** command which cancels the scan of selected endpoints.
2.Fixed some argument types for `xdr-endpoint-scan` command.

## Screenshots
Screen Shot 2021-04-04 at 16 28 15

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

I found a bug in XDR API - When I ran `xdr-endpoint-scan-abort` on multiple endpoints I didn't get an error even though  I ran `xdr-endpoint-scan` only on one of the endpoints (We should get an error on the endpoints that are not being scanned).
On the other hand, when I ran `xdr-endpoint-scan-abort` on only one endpoint and didn't run `xdr-endpoint-scan` before I got an error - so the issue happens only when we send a list of endpoints which some of them are being scanned and some of them not.
@Arsenikr will schedule with them to talk about this bug.

### CODE REVIEW

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 1]
```diff
@@ -2281,6 +2282,62 @@ def endpoint_scan_command(client, args):
...
+        isolate=isolate,
+        hostname=hostname
+    )
+
+    action_id = reply.get("action_id")
+
+    return (
+        tableToMarkdown('Endpoint abort scan', {'Action Id': action_id}, ['Action Id']),
+        {
+            f'{INTEGRATION_CONTEXT_BRAND}.endpointScan.actionId(val.actionId == obj.actionId)': action_id
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2335]: Please consider adding an `Aborted: True` key.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py [hunk 2]
```diff
@@ -956,6 +956,53 @@ def test_endpoint_scan_command_scan_all_endpoints_no_filters_error(requests_mock
...
+
+    client = Client(
+        base_url=f'{XDR_URL}/public_api/v1', headers={}
+    )
+    client._headers = {}
+    markdown, context, raw = endpoint_scan_abort_command(client, test_data['command_args'])
+
+    assert scan_expected_tesult == context
+
+
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:1005]: let's add 1 more case for trying to abort all scans with more args to check the raised error

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 3]
```diff
@@ -2508,3 +2508,57 @@ Initiates a new endpoint script execution kill process.
...
+| platform | Type of operating system. Possible values are: windows, linux, macos, android. | Optional | 
+| alias | Endpoint alias name. | Optional | 
+| isolate | Whether an endpoint has been isolated. Can be "isolated" or "unisolated". Possible values are: isolated, unisolated. | Optional | 
+| hostname | Name of the host. | Optional | 
+| all | Whether to scan all of the endpoints or not. Default is false. Do note that scanning all the endpoints may cause performance issues and latency. Possible values are: true, false. Default is false. | Optional | 
+
+
+#### Context Output
+
+There is no context output for this command.
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:2542]: do not forget to add the scanID and the aborted keys.

**Review by bakatzir — COMMENTED** [review]
Nice clean code. good work.

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md (hunk 3 above)
Comment (barchen1) [reply Packs/CortexXDR/Integrations/CortexXDRIR/README.md]: done

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 4]
```diff
@@ -816,7 +816,7 @@ def endpoint_scan(self, endpoint_id_list=None, dist_name=None, gte_first_seen=No
 
         if alias:
             filters.append({
-                'field': 'alias_name',
+                'field': 'alias',
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:819]: Probably not, in the API docs  it's `alias` and not `alias_name`

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 1 above)
Comment (guykeller) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2335]: @barchen1 Why not two commands with an aborted key?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (guykeller) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:819]: @daryakoval Any input here?

[the remark this answers is not in the document]
Comment (daryakoval) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:819]: @guykeller Hi, I am not related to this command. Didn't added it and didn't used it before.

File: Packs/CortexXDR/ReleaseNotes/3_0_1.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+- Added the ***xdr-endpoint-scan-abort*** command which cancel the scan of selected endpoints.
+- Fixed an issue where ***xdr-endpoint-scan*** command failed using some of it's arguments.
+- Fixed an issue where ***xdr-get-endpoints*** command failed using ``alias`` argument.
```
Comment (bakatzir) [inline Packs/CortexXDR/ReleaseNotes/3_0_1.md:6]:
```suggestion
- Fixed an issue where the ***xdr-get-endpoints*** command failed using the *alias* argument.
```

File: Packs/CortexXDR/pack_metadata.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
     "name": "Palo Alto Networks Cortex XDR - Investigation and Response",
     "description": "Automates Cortex XDR incident response, and includes custom Cortex XDR incident views and layouts to aid analyst investigations.",
     "support": "xsoar",
-    "currentVersion": "3.0.0",
+    "currentVersion": "3.0.1",
```
Comment (bakatzir) [inline Packs/CortexXDR/pack_metadata.json:5]: Can be a minor, 3.1.0. your call.

**Review by bakatzir — COMMENTED** [review]
I'm happy with the changes.
1 last q: have you considered running a scenario of initiating and then aborting a scan in a PB and then check that the `aborted` context key was updated successfully for that scanID?

**Review by guykeller — APPROVED** [review]

**Review by bakatzir — APPROVED** [review]

File: Packs/CortexXDR/ReleaseNotes/3_0_1.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+- Added the ***xdr-endpoint-scan-abort*** command which cancel the scan of selected endpoints.
```
Comment (amshamah419) [inline Packs/CortexXDR/ReleaseNotes/3_0_1.md:4]:
```suggestion
- Added the ***xdr-endpoint-scan-abort*** command which cancels the scan of selected endpoints.
```

File: Packs/CortexXDR/ReleaseNotes/3_0_1.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+- Fixed an issue where ***xdr-endpoint-scan*** command failed using some of it's arguments.
```
Comment (amshamah419) [inline Packs/CortexXDR/ReleaseNotes/3_0_1.md:5]:
```suggestion
- Fixed an issue where the ***xdr-endpoint-scan*** command failed using some of its arguments.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 9]
```diff
@@ -1691,20 +1691,20 @@ script:
...
       description: List of IP addresses.
-      isArray: false
+      isArray: true
       name: ip_list
       required: false
       secret: false
     - default: false
       description: Name of the endpoint group.
-      isArray: false
+      isArray: true
```
Comment (guykeller) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1700]: The name of the group, operating system etc are also lists?
Also, why did we change this?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 10]
```diff
@@ -1751,6 +1751,122 @@ script:
...
+      required: false
+      secret: false
+    - default: false
+      description: Name of the distribution list.
+      isArray: true
+      name: dist_name
+      required: false
+      secret: false
+    - default: false
+      description: Integer in timestamp epoch milliseconds.
```
Comment (guykeller) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1775]: I'd consider rephrasing

Comment (guykeller) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1775]: @amshamah419 Did you look at the YML as well? Maybe I'm missing something here?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 9 above)
Comment (barchen1) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1700]: Yes, it should be a list https://docs.paloaltonetworks.com/cortex/cortex-xdr/cortex-xdr-api/cortex-xdr-apis/response-actions/scan-endpoints.html
We also did it in `get_endpoints_command` function

**Comment by barchen1** [comment]
@amshamah419 Can you please take a look at the yml?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 10 above)
Comment (amshamah419) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1775]: I usually don't unless specified. Regarding this, I would use `Epoch timestamp in milliseconds represented as an integer.`

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 11]
```diff
@@ -1751,6 +1751,122 @@ script:
...
+      description: Name of the host.
+      isArray: true
+      name: hostname
+      required: false
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'false'
+      description: Whether to scan all of the endpoints or not. Default is false.
+        Do note that scanning all the endpoints may cause performance issues and latency.
```
Comment (amshamah419) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1848]:
```suggestion
        Note that scanning all of the endpoints may cause performance issues and latency.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 12]
```diff
@@ -1751,6 +1751,122 @@ script:
...
+      predefined:
+      - 'true'
+      - 'false'
+      required: false
+      secret: false
+    deprecated: false
+    description: Cancel the scan of selected endpoints. A scan can only be aborted
+      if the selected endpoints are in Pending or in Progress status. To scan all
+      endpoints, run this command with argument all=true. Do note that scanning all
+      the endpoints may cause performance issues and latency.
```
Comment (amshamah419) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:1860]:
```suggestion
    description: Cancel the scan of selected endpoints. A scan can only be aborted
      if the selected endpoints are Pending or In Progress. To scan all
      endpoints, run the command with the argument all=true. Note that scanning all
      of the endpoints may cause performance issues and latency.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 13]
```diff
@@ -2508,3 +2508,122 @@ Initiates a new endpoint script execution kill process.
...
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| endpoint_id_list | List of endpoint IDs. | Optional | 
+| dist_name | Name of the distribution list. | Optional | 
+| gte_first_seen | Integer in timestamp epoch milliseconds. | Optional | 
+| gte_last_seen | Integer in timestamp epoch milliseconds. | Optional | 
+| lte_first_seen | Integer in timestamp epoch milliseconds. | Optional | 
+| lte_last_seen | Integer in timestamp epoch milliseconds. | Optional |
```
Comment (amshamah419) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:2531]: Update the docs here to reflect the yml changes.

**Comment by amshamah419** [comment]
@barchen1 YML has been reviewed.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 11 above)
Comment (barchen1) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml]: done

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 10 above)
Comment (barchen1) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml]: done

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 12 above)
Comment (barchen1) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml]: done

**Pull request merged**